src/cell.rs

```diff
@@ -0,0 +1,49 @@
+//! Represent Rokkakari game cell states.
+//!
+//! This module contains various types and functions to represent and manipulate the state of game cells.
+//! A cell is a single hexagon and the smallest unit in a Rokkakari game.
+//! Generally speaking there are two kinds of cells: ***free*** cells and ***wall*** cells.
+//!
+//! # Free Cells
+//!
+//! Free cells are classically painted white in regular Akari.
+//! They have to be lit up and are also the ones on which lamps can be placed.
+//!
+//! # Wall Cells
+//!
+//! Wall cells are classically painted black in regular Akari.
+//! 
+//! - Lamps *cannot* be placed in wall cells.
+//! - Wall cells block off the light from lamps.
+//! - Wall cells can constrain the number of lamps *directly* adjacent to them.
+//!
+//! ## Constaints
+//!
+//! There are two types of wall cells: *constrained* and *unconstrained* ones.
+//! *Constrained* wall cells contain a number from `0` to `3`.
+
+/// The `Cell` type. See [the module level documentation for more](self).
+pub enum Cell {
+    /// A free Rokkakari game cell.
+    /// Its state is represented by [`free::State`](free::State).
+    Free(free::State),
+
+    /// A wall Rokkakari game cell.
+    /// Its state is represented by [`wall::State`](wall::State).
+    Wall(wall::State),
+}
+
+pub mod free {
```

Comment:
Consider moving this submodule to its own file `src/cell/free.rs`, especially when this module grows more.
Replace this `mod` block with `pub mod free;` then.